### ISSUE TITLE
Support iOS Safari Extension

### ISSUE THREAD

**Issue by dploeger** [MaKleSoft/padlock#20]
Common password managers support a custom iOS Safari Extension since iOS 8. That way, a password for the currently visited URL could be fetched from the Vault and copied to the clipboard or even fill out the form.

I will post more information to this issue about the implementation and possible ways in Polymer or cordova.

**Comment by MaKleSoft** [comment]
@dploeger Thanks for the suggestion, but I'm afraid it's a clear 'no' on this one. Auto-filling forms is a feature that I have clearly decided against a while ago and it's unlikely that I will change my mind on this. Please note that Padlock is **not** trying to be a 'common password manager'. The defining property of Padlock is its simplicity and its main feature its *lack* of features. This means that I will be saying 'no' to most feature requests. But please do not be discouraged by this! I appreciate every single suggestion and will consider all of them with an open mind!

**Comment by Zegorax** [comment]
@MaKleSoft Is your comment still valid as of today ? Since Padloc released the browser extension for Chrome and Firefox, it would make sense to support the AutoFill API on iOS as well.
Is a macOS Safari extension planned ?

**Comment by MaKleSoft** [comment]
>  Is your comment still valid as of today ?

Well, yes and no. We're still somewhat sceptical about "auto-fill" as in fully automated form filling as seen in other brother extensions (way too intrusive, major security weakpoint) but I think we've found a pretty good combination of security and convenience with the Chrome and Firefox extensions and with upcoming versions of Safari supporting the Web Extension Spec, there is really no reason not to distribute the extension for Desktop Safari as well...

> Is a macOS Safari extension planned ?

So thats a clear **yes** on that question.

Mobile Safari (and mobile in general) is an entirely different question. iOS and Android actually do offer a pretty solid auto-fill api on the system level and we do want to add support for that (see https://github.com/padloc/padloc/issues/250) but since we're possibly switching to a new native integration layer later this year, there is not much sense in investing a lot of work into that at the moment.

So long story short - yes we do want to offer better auto-fill integration where it can be done securely, but I currently can't give a timeline on that yet. Stay tuned I guess ;)
